src/Microsoft.Health.Api/Features/HealthChecks/CachedHealthCheck.cs

```diff
@@ -139,7 +139,12 @@ private bool HasExpired(DateTimeOffset timestamp)
         => timestamp >= _cache.ExpireTime;
 
     private bool IsUpToDate(DateTimeOffset timestamp)
-        => timestamp < _cache.StaleTime;
+    {
+        // While we will store results whose status is < MinimumCachedHealthStatus, it will effectively
+        // have no caching as IsUpToDate will return false
+        CachedHealthCheckResult currentCache = _cache;
+        return timestamp < currentCache.StaleTime && currentCache.Result.Status >= _options.MinimumCachedHealthStatus;
```

Comment:
Are these values being accessed in a thread safe manner

Reply:
Yes! I am first saving the cache by assigning it to a local variable. The `_cache` variable itself is replaced atomically using `Interlocked.CompareExchange`